drivers/tee/tee_core.c

```diff
@@ -272,27 +272,38 @@ static int params_from_user(struct tee_context *ctx, struct tee_param *params,
 		case TEE_IOCTL_PARAM_ATTR_TYPE_MEMREF_INPUT:
 		case TEE_IOCTL_PARAM_ATTR_TYPE_MEMREF_OUTPUT:
 		case TEE_IOCTL_PARAM_ATTR_TYPE_MEMREF_INOUT:
-			/*
-			 * If we fail to get a pointer to a shared memory
-			 * object (and increase the ref count) from an
-			 * identifier we return an error. All pointers that
-			 * has been added in params have an increased ref
-			 * count. It's the callers responibility to do
-			 * tee_shm_put() on all resolved pointers.
-			 */
-			shm = tee_shm_get_from_id(ctx, ip.c);
-			if (IS_ERR(shm))
-				return PTR_ERR(shm);
+			/* Check for special case of invalid shm id */
+			/*   - corresponds to NULL pointer passed   */
+			if (ip.c != (u64)-1) {
+
+				/*
+				 * If we fail to get a pointer to a shared
+				 * memory object (and increase the ref count)
+				 * from an identifier we return an error. All
+				 * pointers that has been added in params have
+				 * an increased ref count. It's the callers
+				 * responibility to do tee_shm_put() on all
+				 * resolved pointers.
+				 */
+				shm = tee_shm_get_from_id(ctx, ip.c);
+				if (IS_ERR(shm))
+					return PTR_ERR(shm);
+			} else {
+				/* Pass NULL pointer to optee */
+				shm = NULL;
+			}
 
 			/*
 			 * Ensure offset + size does not overflow offset
 			 * and does not overflow the size of the referred
 			 * shared memory object.
 			 */
-			if ((ip.a + ip.b) < ip.a ||
-			    (ip.a + ip.b) > shm->size) {
-				tee_shm_put(shm);
-				return -EINVAL;
+			if (shm) {
+				if ((ip.a + ip.b) < ip.a ||
```

Comment:
This block could follow right after the `return PTR_ERR(shm);` line (row 290) above.

Reply:
True, I'll fix it